hubblestack/extmods/hubble_mods/bexpr.py

```diff
@@ -0,0 +1,317 @@
+# -*- encoding: utf-8 -*-
+"""
+Module to evaluate boolean expressions. Same can be used in both Audit/FDG
+
+Audit Example:
+---------------
+check_unique_id_1:
+  description: 'sample check 1'
+  tag: 'ADOBE-01'
+  implementations:
+    - filter:
+        grains: 'G@osfinger:CentOS*Linux-7'
+      hubble_version: '>3 AND <7 AND <8'
+      module: grep
+      items:
+        - args:
+            path: /etc/ssh/ssh_config
+            pattern: '"^host"'
+            flags: '-E'
+          comparator:
+            type: "string"
+            match: "host*"
+            is_regex: true
+
+check_unique_id_2:
+  description: 'sample check 2'
+  tag: 'ADOBE-02'
+  implementations:
+    - filter:
+        grains: 'G@osfinger:CentOS*Linux-7'
+      hubble_version: '>3 AND <7 AND <8'
+      module: stat
+      items:
+        - args:
+            path: /etc/ssh/ssh_config
+          comparator:
+            type: "dict"
+            match:
+              gid: 0
+              group: "shadow"
+              uid: 0
+              user: "root"
+              mode:
+                type: "file_permission"
+                match:
+                  required_value:  "644"
+                  allow_more_strict: true
+
+check_unique_id:
+  description: 'bexpr check'
+  tag: 'ADOBE-03'
+  implementations:
+    - filter:
+        grains: 'G@osfinger:CentOS*Linux-7'
+      hubble_version: '>3 AND <7 AND <8'
+      module: bexpr
+      items:
+        - args:
+            expr: check_unique_id_1 AND check_unique_id_2
+          comparator:
+            type: boolean
+            match: True
+
+FDG Example:
+------------
+main:
+  description: 'bexpr check'
+  module: bexpr
+  args:
+    expr: check_unique_id_1 AND check_unique_id_2
```

Comment:
FDG module has no mechanism to execute like this. Currently, it moves to next check only via pipe/xpipe etc.
I hope you have tested this